compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/ClassReference.kt

```diff
@@ -144,6 +220,22 @@ public fun FqName.toClassReference(module: ModuleDescriptor): ClassReference {
     ?: throw AnvilCompilationException("Couldn't resolve ClassReference for $this.")
 }
 
+@ExperimentalAnvilApi
+public fun ClassReference.isAnnotatedWith(fqName: FqName): Boolean =
+  annotations.any { it.fqName == fqName }
+
+@ExperimentalAnvilApi
+public fun ClassReference.generateClassName(
+  separator: String = "_"
+): String {
+  return enclosingClassesWithSelf().joinToString(separator = separator) {
+    it.shortName
+  }
+}
+
+@ExperimentalAnvilApi
+public fun ClassReference.asClassName(): ClassName = classId.asClassName()
```

Comment:
The first migrations will be tedious, because we have to unify many of our utilities. But that's the general goal anyway, so it's a win-win :-)